### ISSUE TITLE
requires Java

### ISSUE THREAD

**Issue by Yves-Apel** [IDPF/epubcheck#583]
With the 4.0.0 release, I get always the error message: Cannot check image details (requires Java version 7 or higher).

I have the latest "jre-8u60-macosx-x64.dmg" build installed on my mac 10.10.5

**Comment by rdeltour** [comment]
@Yves-Apel the JRE you installed is probably not configured in the Terminal environment. You can check the version available in the  by running:

```
$ java -version
```

The recommended solution is to install a JDK instead (JDK = Java SE Development Kit). IIRC the Oracle-provided installer should configure the environment for you.

Closing this issue as `invalid`. Feel free to comment here is things don't work.

**Comment by rdeltour** [comment]
In case it helps, I find this [StackOverflow answer](http://stackoverflow.com/a/15625591) best describes the `java` command situation on OS X.

**Comment by Yves-Apel** [comment]
@rdeltour Thanks. JDK worked now. But is everyone expected to di this now on the mac or even also on Windows. Since I wrote 2 books and made a v2b videotraining on ePubs (including the use of epubcheck), I would like to leave an update for my consumers on my blog.

**Comment by rdeltour** [comment]
@Yves-Apel on Mac OS X: yes installing a JDK is the most straightforward way to have Java 7 or higher in the command line environment.

On Windows, the configuration approach is unchanged.

Note that EpubCheck will still work with Java 6, but some images checks are disabled and these info messages will be produced instead.

**Comment by tofi86** [comment]
@Yves-Apel this is due to the change in ownership of the Apple Java from Apple to Oracle. Oracle wouldn't stick to the Apple rules anymore and made the JRE an Internet Plugin only. As per Oracle's JRE [installation document](http://docs.oracle.com/javase/7/docs/webnotes/install/mac/mac-jre.html)

> Installing a JRE from Oracle will not update `java -version` symlinks or add `java` to your path. To be able to do this, you need to install the JDK.

Everyone downloading Java for Mac from www.java.com won't be able to run a java commandline program anymore. To do so, the [JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) is required instead.

----
I will make that clear in Readme and wiki...

**Comment by Yves-Apel** [comment]
@rdeltour @tofi86 Great. I will also update my german followers on this subject. Thanks for your reply.

**Comment by tofi86** [comment]
Updated the Wiki page: https://github.com/IDPF/epubcheck/wiki/Running

**Comment by Yves-Apel** [comment]
@tofi86 Can't wait to get the updated pagina EPUB-Checker when its ready! ;-) Great utility and thanks for your contribution in this.

**Comment by tofi86** [comment]
@Yves-Apel Thanks! You're lucky—I'm on vacation at the moment and will try to get a new version ready today/asap ;-)

**Comment by rdeltour** [comment]
@tofi86 :+1:

**Comment by tofi86** [comment]
@outsideMyBox compiling the JAR for 1.6 with JDK 1.8 should be no problem. This just makes sure that EPubCheck can be run with Java 6 (although it cannot check the image details then). Once you run with Java 7 or later, ImageDetails are working.

Can you check the Java version which is picked up by maven by running those 2 commands:

```
java -version
```

and


```
mvn -v
```

**Comment by outsideMyBox** [comment]
I had checked it (I'm on macOS 10.12):
```
>java -version
java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

> mvn -v
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T17:41:47+01:00)
Maven home: /Users/franck/bin/apache-maven-3.3.9
Java version: 1.8.0_121, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/jre
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "10.12.6", arch: "x86_64", family: "Mac"
```


I use the library in my Scala code so instead of maven I use sbt:
```
sbt -v
[process_args] java_version = '1.8'

# Within sbt I can call java code:
$ eval System.getProperty("java.version")
[info] ans: String = 1.8.0_121
$ eval System.getProperty("java.home")
[info] ans: String = /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/jre
...
```

My JAVA_HOME also seems correct:
`JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home`

It's not blocking so I can live with it but I'd like to know what's happening here :)

**Comment by tofi86** [comment]
Hm, that seems to be correct.

And you get the message `Cannot check image details (requires Java version 7 or higher)` when running the your self-compiled 4.0.2 library from Scala?

Or does it also happen with the 4.0.2 pre-compiled download from GitHub?

Does it also happen when you run epubcheck from the commandline like `java -jar epubcheck-4.0.2.jar your-epub-file.epub` ?

**Comment by outsideMyBox** [comment]
I got it!

"java -jar epubcheck-4.0.2.jar your-epub-file.epub" works well with the downloaded version of epubcheck 4.0.2 from github. 
In my code I use the version from the maven dependency instead (https://mvnrepository.com/artifact/org.idpf/epubcheck/4.0.2). By comparing the dependcies I realised that common-io-3.1.2.jar is missing in the version from maven.
The problem disappears if I include it as an extra dependency.
